### PR TITLE
Don't exceed max allowed color attachments and fix get proc addr names.

### PR DESCRIPTION
**Pull request by lentinem**

### CODE REVIEW

File: layers/device_limits.cpp [hunk 1]
```diff
@@ -676,11 +699,13 @@ VK_LAYER_EXPORT VKAPI_ATTR PFN_vkVoidFunction VKAPI_CALL vkGetDeviceProcAddr(VkD
         return (PFN_vkVoidFunction)vkDestroyDevice;
     if (!strcmp(funcName, "vkGetDeviceQueue"))
         return (PFN_vkVoidFunction)vkGetDeviceQueue;
-    if (!strcmp(funcName, "CreateCommandPool"))
+    if (!strcmp(funcName, "vkCreateRenderPass"))
+        return (PFN_vkVoidFunction)vkCreateRenderPass;
+    if (!strcmp(funcName, "vkCreateCommandPool"))
```
Comment (chrisforbes) [inline layers/device_limits.cpp]: This looks like a fix for another issue. At least call it out in the commit message?

File: layers/device_limits.cpp [hunk 2]
```diff
@@ -468,6 +468,29 @@ VK_LAYER_EXPORT VKAPI_ATTR void VKAPI_CALL vkDestroyDevice(VkDevice device, cons
     layer_data_map.erase(key);
 }
 
+VK_LAYER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL vkCreateRenderPass(VkDevice device, const VkRenderPassCreateInfo *pCreateInfo,
+                                                                  const VkAllocationCallbacks *pAllocator,
+                                                                  VkRenderPass *pRenderPass) {
+    layer_data *dev_data = get_my_data_ptr(get_dispatch_key(device), layer_data_map);
+    bool skip_call = false;
+    if (dev_data) {
```
Comment (chrisforbes) [inline layers/device_limits.cpp]: Do you need to test for this? Things are /deeply/ wrong if you don't have layer data for the device; elsewhere we don't check.

File: layers/device_limits.cpp [hunk 3]
```diff
@@ -468,6 +468,29 @@ VK_LAYER_EXPORT VKAPI_ATTR void VKAPI_CALL vkDestroyDevice(VkDevice device, cons
...
+                                     reinterpret_cast<uint64_t>(device), __LINE__, DEVLIMITS_INVALID_ATTACHMENT_COUNT, "DL",
+                                     "Cannot create a render pass with %d color attachments. Max is %d.",
+                                     pCreateInfo->pSubpasses[i].colorAttachmentCount, max_color_attachments);
+            }
+        }
+    }
+    if (skip_call) {
+        return VK_ERROR_VALIDATION_FAILED_EXT;
+    }
+    return get_my_data_ptr(get_dispatch_key(device), layer_data_map)
```
Comment (chrisforbes) [inline layers/device_limits.cpp]: `return dev_data->device_dispatch_table->CreateRenderPass(...`

File: layers/device_limits.cpp (hunk 2 above)
Comment (lentinem) [inline layers/device_limits.cpp]: I could remove it; I'm not too picky on this point so whichever is fine with me.

Comment (chrisforbes) [inline layers/device_limits.cpp]: Well, despite the check you're about to segfault on :490 in this case so I'd just remove it.

File: layers/device_limits.cpp [hunk 4]
```diff
@@ -468,6 +468,28 @@ VK_LAYER_EXPORT VKAPI_ATTR void VKAPI_CALL vkDestroyDevice(VkDevice device, cons
...
 }
 
+VK_LAYER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL vkCreateRenderPass(VkDevice device, const VkRenderPassCreateInfo *pCreateInfo,
+                                                                  const VkAllocationCallbacks *pAllocator,
+                                                                  VkRenderPass *pRenderPass) {
+    layer_data *dev_data = get_my_data_ptr(get_dispatch_key(device), layer_data_map);
+    bool skip_call = false;
+    if (dev_data) {
+        uint32_t max_color_attachments = dev_data->physicalDeviceProperties.limits.maxColorAttachments;
+        for (uint32_t i = 0; i <= pCreateInfo->subpassCount; ++i) {
```
Comment (chrisforbes) [inline layers/device_limits.cpp]: `<`. You're reading off the end.

**Comment by chrisforbes** [comment]
:+1: with those issues fixed.

**Comment by tobine** [comment]
cherry-picked and pushed to master

**Pull request closed without merging**